contrib/meson/meson/lib/meson.build

```diff
@@ -1,57 +1,81 @@
 # #############################################################################
-# Copyright (c) 2018-present    lzutao <taolzu(at)gmail.com>
+# Copyright (c) 2018-present        lzutao <taolzu(at)gmail.com>
+# Copyright (c) 2022-present        Tristan Partin <tristan(at)partin.io>
 # All rights reserved.
 #
 # This source code is licensed under both the BSD-style license (found in the
 # LICENSE file in the root directory of this source tree) and the GPLv2 (found
 # in the COPYING file in the root directory of this source tree).
 # #############################################################################
 
-lz4_root_dir = '../../../..'
-
-liblz4_includes = [include_directories(join_paths(lz4_root_dir, 'lib'))]
-liblz4_sources = [join_paths(lz4_root_dir, 'lib/lz4.c'),
-  join_paths(lz4_root_dir, 'lib/lz4frame.c'),
-  join_paths(lz4_root_dir, 'lib/lz4hc.c'),
-  join_paths(lz4_root_dir, 'lib/xxhash.c')]
-liblz4_c_args = []
-
-liblz4_debug_cflags = []
-if use_debug
-  liblz4_c_args += '-DLZ4_DEBUG=@0@'.format(debug_level)
-  if [compiler_gcc, compiler_clang].contains(cc_id)
-    liblz4_debug_cflags = ['-Wextra', '-Wcast-qual', '-Wcast-align', '-Wshadow',
-      '-Wswitch-enum', '-Wdeclaration-after-statement', '-Wstrict-prototypes',
-      '-Wundef', '-Wpointer-arith', '-Wstrict-aliasing=1']
-  endif
+sources = files(
+  lz4_source_root / 'lib/lz4.c',
+  lz4_source_root / 'lib/lz4frame.c',
+  lz4_source_root / 'lib/lz4hc.c',
+  lz4_source_root / 'lib/xxhash.c'
+)
+
+c_args = []
+
+compile_args = []
+
+if host_machine.system() == 'windows' and default_library != 'static'
+  c_args += '-DLZ4_DLL_EXPORT=1'
+endif
+
+if get_option('memory-usage') > 0
+  compile_args += '-DLZ4_MEMORY_USAGE=@0@'.format(get_option('memory-usage'))
```

Comment:
You set this elsewhere using add_project_arguments.

Reply:
`compile_args` is exported in `declare_dependency(compile_args:)`. Not used for c_args. Unless you had a different intention with your comment?

Reply:
No, I just mistook what its purpose was here. Sorry!

...

Does that mean this needs to be in the generated pkgconfig file too?

Reply:
@Cyan4973 would be the authority here. Maybe the behavior I wrote is not what should be exhibited.

Reply:
Note that I'm not familiar with `meson`.
Consequently, I'm unaware of the consequences of employing `compile_args`.
From a `Makefile` perspective, something like `-DLZ4_MEMORY_USAGE=#` is supposed to belong to the `CPPFLAGS` variable, rather than the `CFLAGS` one. The difference can be substantial since `CFLAGS` are ignored at link stage, but `CPPFLAGS` are not. It's unclear if this level of differentiation exists at `meson` level.

Then, if, as mentioned by @eli-schwartz, the flags are then also present in `pkgconfig`, well this is unusual, but it seems it would allow the library to be compiled with a different hash table size than the default size (directly defined in the source code). This is pretty advanced stuff, that is rarely ever used. Furthermore, in most cases, this size is abstracted away and does not matter. So it takes a very advanced scenario for this case to matter. And such advanced scenario may more likely happen in the embedded world, where a `pkgconfig` is less likely to be a thing.
So, to summarize, yes, in some rare cases, it could prove useful. But as importantly, in all cases I can think of, and presuming that the `pkgconfig` remains in sync with the installed library, it should not be detrimental.

Reply:
`compile_args` tells a project using this lz4 dep to compile with the listed `compile_args` in the compiler invocation. Based on what you said this isn't expected nor popular. Therefore I will remove it!